### PR TITLE
adds anvils

### PR DESCRIPTION
**Pull request by MadmanMartian**
- they go clang if thrown at somebody
 - Need to be strong to pull them, need to be strong and hulkster to pick it up to throw it
 - can be used for crafting
 - misc bug fixes related to dorf crafting

🆑 
* rscadd: Adds anvils. Ordered through cargo.

### CODE REVIEW

File: code/game/objects/items/anvil.dm [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	impactsound = 'sound/misc/clang.ogg'
+	flags = FPRINT | TWOHANDABLE | MUSTTWOHAND
+	density = 1
+	throwforce = 40
+
+/obj/item/anvil/can_pickup(mob/living/M)
+	if(!..())
+		return FALSE
+	if((M_HULK in M.mutations) && (M_STRONG in M.mutations))
+		M.visible_message("<span class = 'warning'>\The [M] attempts to pick up \the [src].</span>")
```
Comment (DamianX) [inline code/game/objects/items/anvil.dm]: \>can_pickup
\>contains a `visible_message`
i dislike this

**Comment by ghost** [comment]
Check that it can't be moved by a light breeze

And also I guess you can move it via locker

Probably should be a structure tbh

**Comment by MadmanMartian** [comment]
It takes a bit of oomph to push it now. Can't move it via locker as it's a dense object.

File: code/game/objects/items/anvil.dm [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	icon_state = "anvil"
+	impactsound = 'sound/misc/clang.ogg'
+	flags = FPRINT | TWOHANDABLE | MUSTTWOHAND
+	density = 1
+	throwforce = 40
+
+/obj/item/anvil/can_pickup(mob/living/M)
+	if(!..())
+		return FALSE
+	if((M_HULK in M.mutations) && (M_STRONG in M.mutations))
```
Comment (unid15) [inline code/game/objects/items/anvil.dm]: https://github.com/vgstation-coders/vgstation13/blob/Bleeding-Edge/code/modules/mob/living/helper_procs.dm#L31

Use ``M.get_strength()`` for all checks in this file

**Review by unid15 — CHANGES_REQUESTED** [review]

File: code/game/objects/items/stacks/stack_recipes.dm [hunk 3]
```diff
@@ -226,7 +226,8 @@ var/list/datum/stack_recipe/metal_recipes = list (
 	null,
 	new/datum/stack_recipe("iron door", /obj/machinery/door/mineral/iron, 					20, 			one_per_turf = 1, on_floor = 1),
 	new/datum/stack_recipe("stove", /obj/machinery/space_heater/campfire/stove, 			5, time = 25, 	one_per_turf = 1, on_floor = 1),
-	new/datum/stack_recipe/dorf("training sword", /obj/item/weapon/melee/training_sword,	4, time = 12,	on_floor = 1, inherit_material = TRUE)
+		new/datum/stack_recipe/dorf("training sword", /obj/item/weapon/melee/training_sword,	4, time = 12,	on_floor = 1, inherit_material = TRUE),
```
Comment (jknpj) [inline code/game/objects/items/stacks/stack_recipes.dm]: indent

**Comment by MadmanMartian** [comment]
Removed it just being a proximity thing for crafting - We'll not go the cheap and casual way with this.

Also can't just magic one out of thin air using plasteel and metal, you'll have to order one for the time being.

**Comment by MusgardTiger** [comment]
Can't we add it in a corner in some maintenance tunnel?

**Comment by SonixApache** [comment]
I can't figure it out. Does this actually do anything?

**Comment by MadmanMartian** [comment]
No

**Pull request merged**